### PR TITLE
Collected small fixes and updates

### LINKED ISSUE #4016: [BUG] Documentation-Code Mismatch for Default Error Argument in fix halt
**Summary**

Discrepancy between `fix halt` documentation and default optional argument in the code.

**LAMMPS Version and Platform**
LAMMPS (23 Jun 2022 - Update 1)  

**Expected Behavior**

The `fix halt` [documentation](https://docs.lammps.org/fix_halt.html#default) states that the default error argument is hard, but the code in [`fix_halt.cpp` ](src/fix_halt.cpp)at line 92 sets `eflag = SOFT` as the default.

### LINKED ISSUE #4033: [BUG] OpenMP: omp.h not found by CMake when using oneapi preset
**Summary**

OpenMP detection fails with cmake when using the oneapi.cmake preset, because it has two openmp flags: -qopenmp -qopenmp-simd
This causes the flags to get quoted in the detection script, which makes it fail.
Therefore, CMake always fails the OpenMP test, and turns the flags off, even when it is available.

**LAMMPS Version and Platform**

Tested with latest stable version of 2Aug2023_update2 and the latest develop branch.
Tested with CMake versions 3.20.4 and 3.28.1.
Tested with Intel oneapi versions 2023.0.0 and 2024.0.2.
Tested with RHEL 7.5 and openSUSE Leap 15.5.

**Expected Behavior**

As Intel oneapi has OpenMP capabilities, the CMake script should have detected it, as in the following log:
> -- Looking for C++ include omp.h 
> -- Looking for C++ include omp.h - found

**Actual Behavior**

Currently, CMake doesn’t detect the OpenMP when used with the oneapi preset, which turn off OpenMP capabilities:
> -- Looking for C++ include omp.h
> -- Looking for C++ include omp.h - not found

**Steps to Reproduce**

Install the Intel HPC kit.
```
cd <LAMMPS folder>
mkdir test
cd test
cmake -C ../cmake/presets/oneapi.cmake ../cmake
```

**Further Information, Files, and Links**

The reason is that the two OpenMP flags are used quoted in the test script, resulting in an unknown argument error, as seen in the log:

> Change Dir: /home/test_user/rvc/lammps/test/CMakeFiles/CMakeTmp
> 
> Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_e55b4/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_e55b4.dir/build.make CMakeFiles/cmTC_e55b4.dir/build
> gmake[1]: Entering directory '/home/test_user/rvc/lammps/test/CMakeFiles/CMakeTmp'
> Building CXX object CMakeFiles/cmTC_e55b4.dir/CheckIncludeFile.cxx.o
> /opt/intel/oneapi/compiler/2024.0/bin/icpx   -qopenmp -qopenmp-simd -fPIE -std=c++11 -MD -MT CMakeFiles/cmTC_e55b4.dir/CheckIncludeFile.cxx.o -MF CMakeFiles/cmTC_e55b4.dir/CheckIncludeFile.cxx.o.d -o CMakeFiles/cmTC_e55b4.dir/CheckIncludeFile.cxx.o -c /home/test_user/rvc/lammps/test/CMakeFiles/CMakeTmp/CheckIncludeFile.cxx
> Linking CXX executable cmTC_e55b4
> /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_e55b4.dir/link.txt --verbose=1
> /opt/intel/oneapi/compiler/2024.0/bin/icpx  -qopenmp -qopenmp-simd "-qopenmp -qopenmp-simd" CMakeFiles/cmTC_e55b4.dir/CheckIncludeFile.cxx.o -o cmTC_e55b4  -liomp5
> icpx: error: unknown argument: '-qopenmp -qopenmp-simd'
> gmake[1]: *** [CMakeFiles/cmTC_e55b4.dir/build.make:100: cmTC_e55b4] Error 1
> gmake[1]: Leaving directory '/home/test_user/rvc/lammps/test/CMakeFiles/CMakeTmp'
> gmake: *** [Makefile:127: cmTC_e55b4/fast] Error 2

It’s possible to make the detection work if you force a single OpenMP flag:

> cmake -C ../cmake/presets/oneapi.cmake -DOpenMP_CXX_FLAGS=-qopenmp ../cmake

This might be because the following omp.h detection function in LAMMPSUnits.cmake only expects one OpenMP flag, but this is beyond my area of expertise:
```
# helper function to check for usable omp.h header
function(check_omp_h_include)
  find_package(OpenMP COMPONENTS CXX QUIET)
  if(OpenMP_CXX_FOUND)
    set(CMAKE_REQUIRED_FLAGS ${OpenMP_CXX_FLAGS})
    set(CMAKE_REQUIRED_INCLUDES ${OpenMP_CXX_INCLUDE_DIRS})
    set(CMAKE_REQUIRED_LINK_OPTIONS ${OpenMP_CXX_FLAGS})
    set(CMAKE_REQUIRED_LIBRARIES ${OpenMP_CXX_LIBRARIES})
    check_include_file_cxx(omp.h _have_omp_h)
  else()
    set(_have_omp_h FALSE)
  endif()
  set(HAVE_OMP_H_INCLUDE ${_have_omp_h} PARENT_SCOPE)
endfunction()

```

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request combines multiple small fixes and updates that do not warrant a pull request of their own.

**Related Issue(s)**

Fixes #4016 
Closes #4024 
Fixes #4033 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No known issues.

**Implementation Notes**

The following individual changes are included:
- move of "disk_free()" function from fix halt to platform namespace and port to Windows
- correct documentation of fix halt match the code
- update and clarify discussion of LAMMPS versions and releases
- modernize fix halt sources
- fix small omission in msi2lmp in warning message
- address some minor issues reported by coverity scan for compute slcsa/atom, compute pace, compute sna/atom
- remove unused class members and commented out debug code from compute pace
- silence some compiler warnings
- small enhancements signaled by clang-tidy
- let ReaxFF force field parser cleanly handle files without hydrogen bond parameters
- fix uninitialized data access bug in Grid2d and Grid3d classes
- skip python tests requiring numpy on macOS as those seem to fail randomly with GitHub actions with numpy installed (but not detected)
- check for using INTEL package kspace with run style `verlet/split`. If yes, error out with message that they are incompatible
- improve compatibility of openapi.cmake preset

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.

### CODE REVIEW

File: doc/github-development-workflow.md
Comment (doronbehar) [inline doc/github-development-workflow.md:40]: `associated` tags is not clear IMO. I assume you mean `patch_` and `stable_` tags? Maybe you should document all this in 1 place?

File: doc/src/Howto_github.rst
Comment (doronbehar) [inline doc/src/Howto_github.rst:480]: Once again I think that having all of this information scattered in different docs is not ideal...

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:51]: A branch name is a word you view in your terminal, hence I'd suggest:

```suggestion
``develop`` branch of the git repository after they pass automated testing
```

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:58]: Same here

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:60]: And here for ``release``.

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:78]:
```suggestion
branches are updated and two tags are applied, with tags such as ``patch_1May2014`` and ``stable_1May2014``, in conjunction.
```

Or perhaps:

```suggestion
branches are updated and two tags are applied, with the prefixes ``patch_`` and ``stable_``, in conjunction.
```

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:88]: This page seems much better altogether. I still think that a table such as I suggested in https://github.com/lammps/lammps/pull/4024 as a sort of summary would be helpful for the lazy readers. I'd even put it in the beginning of the file, as a sort of TL;DR.

File: doc/src/Manual_version.rst
Comment (doronbehar) [inline doc/src/Manual_version.rst:84]: Again branch names should be `<code>` formatted IMO.

File: doc/github-development-workflow.md
Comment (akohlmey) [reply doc/github-development-workflow.md:40]: This is not a file meant to be read by regular uses, but only by LAMMPS developers. Those know what the tags mean.

Documenting "everything in one place" is not a good idea since there are different "consumers" of the documentation: regular users, users that make modifications to the source, LAMMPS developers. In particular for the first kind, there needs to be some redundancy, since those rarely follow links and thus some minimal information has to be put right where they need it.

File: doc/src/Howto_github.rst
Comment (akohlmey) [reply doc/src/Howto_github.rst:480]: Neither is having everything combined where fewer people will see it.

File: doc/src/Manual_version.rst
Comment (akohlmey) [reply doc/src/Manual_version.rst:88]: @doronbehar thanks for your feedback. Please understand that your personal preferences (and frustrations) are rather different from the consensus among the LAMMPS developers. Over the years there must have been many tens of thousands of people that were not having the kind of problems that you had. Thus if we make too drastic changes, the effect may be counterproductive. It is always a struggle between providing complete and detailed information and providing just sufficient information without distracting people too much. For content at the beginning of the manual, the preference is for the latter and I am concerned that the changes I am proposing are already pushing too far. But that is why we have a require review from at least one other LAMMPS developer before changes are applied and I am waiting for that kind of feedback before changes will be included in the distribution.

**Review by stanmoore1 — APPROVED** [review]
I approve

**Review by ndtrung81 — APPROVED** [review]

**Review by sjplimp — APPROVED** [review]

**Pull request merged**
